### PR TITLE
support equal and not_equal in topi

### PR DESCRIPTION
**Pull request by liangfu**
Add support to `count_nonzero` in topi, works well for llvm. However, current cuda schudule for reduction don't work well with the proposed operator.

### CODE REVIEW

**Comment by tqchen** [comment]
Can we compose this operator by having a equal operator then followed by a sum, if so, we should support equal and instead of this function

**Comment by liangfu** [comment]
I gave it a try, and test cases has been added.

**Comment by tqchen** [comment]
what i mean instead is that we should not support count nonzero function, but instead support topi.equal and topi.not_equal, so we can compose count nonzero by equal then astype(int32) and the sum

**Comment by liangfu** [comment]
i could do that, but my original intention is to estimate buffer size for topi.nonzero, which could be useful for converting dense matrices into sparse ones.

**Comment by tqchen** [comment]
I agree, nevertheless these can be done in the composed fashion so we can express the same thing but much more flexibily

**Comment by liangfu** [comment]
Okay, here is another update.
I think it's fine if we don't need `count_nonzero` in `topi`, but how about a `topi.nonzero`?
It could be a progressive step towards an introduction to sparse tensor.

File: topi/tests/python/test_topi_broadcast.py [hunk 1]
```diff
@@ -79,6 +79,9 @@ def check_device(device):
         out_npy = fnumpy(lhs_npy, rhs_npy)
         out_nd = tvm.nd.array(np.empty(out_npy.shape).astype(C.dtype), ctx)
         foo(lhs_nd, rhs_nd, out_nd)
+        if fnumpy==np.equal:
```
Comment (tqchen) [inline topi/tests/python/test_topi_broadcast.py]: remove the print here

**Comment by tqchen** [comment]
I agree that topi.nonzero is useful and can be supported in a separate PR. although it is unclear if we need additional gather primitive/hybrid script to support this operation

**Pull request merged**